src/Microsoft.AspNetCore.SignalR.Common/Internal/Formatters/BinaryMessageFormatter.cs

```diff
@@ -2,21 +2,23 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
-using System.Buffers;
 using System.IO;
 
 namespace Microsoft.AspNetCore.SignalR.Internal.Formatters
 {
     public static class BinaryMessageFormatter
     {
-        public unsafe static void WriteMessage(ReadOnlySpan<byte> payload, Stream output)
+        public static void WriteLengthPrefix(long length, Stream output)
         {
             // This code writes length prefix of the message as a VarInt. Read the comment in
             // the BinaryMessageParser.TryParseMessage for details.
 
-            var lenBuffer = stackalloc byte[5];
+#if NETCOREAPP2_1
+            Span<byte> lenBuffer = stackalloc byte[5];
+#else
+            var lenBuffer = new byte[5];
```

Comment:
Why isn't this stackalloc?

Reply:
Unsafe?

Reply:
Also, does `Stream` have a `WriteAsync` that takes `byte*`? We used to copy to a pooled array before, which seems unnecessary.